docs/user-guide/a1-01-matrix-layout.md

```diff
@@ -33,37 +57,40 @@ slang::ISession* session;
 globalSession->createSession(slangSessionDesc, &session);
 ```
 
-This make make Slang treat all matrices as in column-major layout, and emit `column_major` qualifier in resulting code.
+This makes Slang treat all matrices as in `column-major` layout, and for example emitting `column_major` qualifier in resulting HLSL code.
+
+Alternatively the default layout can be set via
 
-Note that if you choose to use column-major layout, you either need to flip the matrix multiplication order in shader code or fill in the matrix in transpose order in host code.
+* `setMatrixLayoutMode`/`spSetMatrixLayoutMode` API calls
+* `-matrix-layout-row-major` or `-matrix-layout-column-major` command line options
+  * or via `spProcessCommandLineArguments`/`processCommandLineArguments`
 
-Summary
--------------
+Examples
+--------
 
-In summary, we put together all options you have to ensure correct result:
+There are 4 variables:
 
-**Option 1: using row-major matrix layout, and `mul(m, v)` math convention**
+* Host vector interpretation (row or column) - and therefore effective tranform order (column) `m * v` or (row) `v * m`
+* Host matrix memory layout
```

Comment:
I am not sure if it makes sense to separate matrix memory layout to host and shader, because it only makes logical sense if host and shader assumes the same layout, no?

Reply:
I think it's a useful distinction - especially if we don't have the ability to just set the convention, as is the case for C++/CUDA. That as is discussed now host matrix layout is not arbitrary if performance is a concern.